### PR TITLE
Updates to reflect the incoming changes to setting the preview language

### PR DESCRIPTION
**Pull request by staubina**
## [TNL-4742](https://openedx.atlassian.net/browse/TNL-4742)

Updated the internationalization developer documentation to reflect the upcoming changes for setting the preview language code.
### Date Needed (optional)

Release date is TBD
### Reviewers

Possible roles follow. The PR submitter checks the boxes after each reviewer finishes and gives :+1:. 
- [x] Subject matter expert: @cahrens 
- [x] Subject matter expert: @nedbat 
- [x] Doc team review (sanity check/copy edit/dev edit): @catong @lamagnifica @pdesjardins @srpearce
- [x] Product review: @sstack22 
### Testing
- [x] Ran ./run_tests.sh without warnings or errors
### Sandbox (optional)
- [x] Point to or build a sandbox for the software change and add a link here
- Links for testing
  - https://staubina.sandbox.edx.org/update_lang/
  - https://studio-staubina.sandbox.edx.org/update_lang/
### Post-review
- [x] Add a comment with the description of this change or link this PR to the next release notes task.
- [x] Squash commits

### CODE REVIEW

**Comment by staubina** [comment]
I will add a Sandbox tomorrow after checking in on how to create one for documentation.

File: en_us/developers/source/conventions/internationalization/i18n.rst [hunk 1]
```diff
@@ -408,8 +408,8 @@ Next, restart the LMS and Studio to load in the new translation files. ::
     $ paver devstack lms
     $ paver devstack studio
 
-When you visit Studio or the LMS, append ``?preview-lang=eo`` to the URL to
-view the webpage in the dummy testing language.
+Append ``/update_lang/`` to the root LMS or Studio URL and use the form to set 
+the preview language to ``eo`` the test language.
```
Comment (cahrens) [inline en_us/developers/source/conventions/internationalization/i18n.rst]: I find the wording of this to be awkward. How about-- 

Append ... and use the form to set the preview language. The language code `eo` can be used to specify the test language.

File: en_us/developers/source/conventions/internationalization/i18n.rst [hunk 2]
```diff
@@ -440,16 +440,32 @@ page layout, such as columns that do not fit, or text that is truncated (the
...
-URL to reset your session to your base language.
+When you have finished reviewing, append ``/update_lang/`` to the LMS or 
+Studio URL and use the form to reset your session to your base language.
+
+Set Preview Language
+====================
+
+To set or clear the preview language append ``/update_lang/`` to the root LMS or 
+Studio URL. If you are logged in you will be presented with a form to set or 
+clear the preview language. Set the Language code (for example use eo for the test
```
Comment (cahrens) [inline en_us/developers/source/conventions/internationalization/i18n.rst]: `eo`

File: en_us/developers/source/conventions/internationalization/i18n.rst [hunk 3]
```diff
@@ -440,16 +440,32 @@ page layout, such as columns that do not fit, or text that is truncated (the
...
+Studio URL and use the form to reset your session to your base language.
+
+Set Preview Language
+====================
+
+To set or clear the preview language append ``/update_lang/`` to the root LMS or 
+Studio URL. If you are logged in you will be presented with a form to set or 
+clear the preview language. Set the Language code (for example use eo for the test 
+language Esperanto) and select the "Submit" button to set the preview language. 
+Use the "Reset" button to reset the preview language to the user's default setting
```
Comment (cahrens) [inline en_us/developers/source/conventions/internationalization/i18n.rst]: Missing a period.

File: en_us/developers/source/conventions/internationalization/i18n.rst [hunk 4]
```diff
@@ -440,16 +440,32 @@ page layout, such as columns that do not fit, or text that is truncated (the
...
+
+Notes: 
+#. This setting will require a refresh of the page to take effect and will 
+persist with the user's session
```
Comment (cahrens) [inline en_us/developers/source/conventions/internationalization/i18n.rst]: Missing a period.

File: en_us/developers/source/conventions/internationalization/i18n.rst [hunk 5]
```diff
@@ -440,16 +440,32 @@ page layout, such as columns that do not fit, or text that is truncated (the
...
+
+#. Login is not required to set the preview language.
```
Comment (cahrens) [inline en_us/developers/source/conventions/internationalization/i18n.rst]: Really? I thought login was required.

Comment (staubina) [inline en_us/developers/source/conventions/internationalization/i18n.rst]: wow... major typo there... that should be now...

File: en_us/developers/source/conventions/internationalization/i18n.rst [hunk 6]
```diff
@@ -408,8 +408,9 @@ Next, restart the LMS and Studio to load in the new translation files. ::
...
     $ paver devstack studio
 
-When you visit Studio or the LMS, append ``?preview-lang=eo`` to the URL to
-view the webpage in the dummy testing language.
+Append ``/update_lang/`` to the root LMS or Studio URL and use the form to set 
+the preview language. The language code ``eo`` can be used to specify the test 
+language.
 
 Instead of plain English strings, you should see specially-accented English
 strings that look like this example.
```
Comment (lamagnifica) [inline en_us/developers/source/conventions/internationalization/i18n.rst:416]: @staubina would you mind fixing the error on line 424, by changing "following" to "follow"?

File: en_us/developers/source/conventions/internationalization/i18n.rst [hunk 7]
```diff
@@ -440,16 +441,33 @@ page layout, such as columns that do not fit, or text that is truncated (the
...
-user interface is properly "flipped" to right-to-left mode. Note that certain
-page elements might not look correct because they are controlled by the
-browser. For the best testing, switch your browser's language to Arabic or
-another RTL language (Hebrew, Persian, or Urdu) as well. See our `RTL UI
-Guidelines`_ for information about fixing any issues that you find.
+Finally, append ``/update_lang/`` to the root LMS or Studio URL and set the 
+language code to ``rtl`` to view your feature in the dummy right-to-left 
+(RTL) language. Test to make sure that the user interface is properly 
+"flipped" to right-to-left mode. Note that certain page elements might not 
+look correct because they are controlled by the browser. For the best testing,
```
Comment (lamagnifica) [inline en_us/developers/source/conventions/internationalization/i18n.rst]: the best --> more effective

File: en_us/developers/source/conventions/internationalization/i18n.rst [hunk 8]
```diff
@@ -440,16 +441,33 @@ page layout, such as columns that do not fit, or text that is truncated (the
...
 
-When you have finished reviewing, append ``?clear-lang`` to the LMS or Studio
-URL to reset your session to your base language.
+When you have finished reviewing, append ``/update_lang/`` to the LMS or 
+Studio URL and reset your session to your base language.
+
+Set Preview Language
+====================
+
+To set or clear the preview language append ``/update_lang/`` to the root LMS or
```
Comment (lamagnifica) [inline en_us/developers/source/conventions/internationalization/i18n.rst]: add a comma after language

File: en_us/developers/source/conventions/internationalization/i18n.rst [hunk 9]
```diff
@@ -440,16 +441,33 @@ page layout, such as columns that do not fit, or text that is truncated (the
...
+When you have finished reviewing, append ``/update_lang/`` to the LMS or 
+Studio URL and reset your session to your base language.
+
+Set Preview Language
+====================
+
+To set or clear the preview language append ``/update_lang/`` to the root LMS or 
+Studio URL. If you are logged in you will be presented with a form to set or 
+clear the preview language. Set the Language code (for example use ``eo`` for the test 
+language Esperanto) and select the "Submit" button to set the preview language.
```
Comment (lamagnifica) [inline en_us/developers/source/conventions/internationalization/i18n.rst]: and select the "Submit" button --> , and then select Submit (where Submit is bold)

File: en_us/developers/source/conventions/internationalization/i18n.rst [hunk 10]
```diff
@@ -440,16 +441,33 @@ page layout, such as columns that do not fit, or text that is truncated (the
...
+Studio URL and reset your session to your base language.
+
+Set Preview Language
+====================
+
+To set or clear the preview language append ``/update_lang/`` to the root LMS or 
+Studio URL. If you are logged in you will be presented with a form to set or 
+clear the preview language. Set the Language code (for example use ``eo`` for the test 
+language Esperanto) and select the "Submit" button to set the preview language. 
+Use the "Reset" button to reset the preview language to the user's default setting.
```
Comment (lamagnifica) [inline en_us/developers/source/conventions/internationalization/i18n.rst]: replace the quotation marks with two asterisks

File: en_us/developers/source/conventions/internationalization/i18n.rst [hunk 11]
```diff
@@ -440,16 +441,33 @@ page layout, such as columns that do not fit, or text that is truncated (the
...
+
+Notes:
```
Comment (lamagnifica) [inline en_us/developers/source/conventions/internationalization/i18n.rst]: to place the notes into a colored block with a bolded title, use this format:

.. note::
  #. words and words
  #. words and words

File: en_us/developers/source/conventions/internationalization/i18n.rst [hunk 12]
```diff
@@ -440,16 +441,33 @@ page layout, such as columns that do not fit, or text that is truncated (the
...
-When you have finished reviewing, append ``?clear-lang`` to the LMS or Studio
-URL to reset your session to your base language.
+When you have finished reviewing, append ``/update_lang/`` to the LMS or 
+Studio URL and reset your session to your base language.
+
+Set Preview Language
+====================
+
+To set or clear the preview language append ``/update_lang/`` to the root LMS or 
+Studio URL. If you are logged in you will be presented with a form to set or
```
Comment (lamagnifica) [inline en_us/developers/source/conventions/internationalization/i18n.rst]: the very last thing on the page essentially says that you must be logged in. I suggest that you rephrase and resequence  this introduction to put sign in as a prerequisite. 
Before you set the preview language, sign in to either the LMS or Studio. Then, append ... URL. A form appears for you to set or clear...

File: en_us/developers/source/conventions/internationalization/i18n.rst [hunk 13]
```diff
@@ -440,16 +441,33 @@ page layout, such as columns that do not fit, or text that is truncated (the
...
+To set or clear the preview language append ``/update_lang/`` to the root LMS or 
+Studio URL. If you are logged in you will be presented with a form to set or 
+clear the preview language. Set the Language code (for example use ``eo`` for the test 
+language Esperanto) and select the "Submit" button to set the preview language. 
+Use the "Reset" button to reset the preview language to the user's default setting.
+
+Notes: 
+
+#. This setting will require a refresh of the page to take effect and will persist 
+   with the user's session.
```
Comment (lamagnifica) [inline en_us/developers/source/conventions/internationalization/i18n.rst]: with the user's --> for your

File: en_us/developers/source/conventions/internationalization/i18n.rst [hunk 14]
```diff
@@ -440,16 +441,33 @@ page layout, such as columns that do not fit, or text that is truncated (the
...
+
+#. Login is required to set the preview language.
```
Comment (lamagnifica) [inline en_us/developers/source/conventions/internationalization/i18n.rst]: if you add sign in as a prerequisite, you won't need the note to go to a numbered list

File: en_us/developers/source/conventions/internationalization/i18n.rst (hunk 13 above)
Comment (lamagnifica) [inline en_us/developers/source/conventions/internationalization/i18n.rst]: actually, this, to me, reads like the final step of the procedure. Maybe say,
Refresh your browser page to display the page in the selected language. The language persists for the duration of your session.

**Comment by staubina** [comment]
@lamagnifica 
I updated the PR per your comments.

**Comment by lamagnifica** [comment]
Thanks @staubina!  I made some copy edits.  
Depending on your availability and level of interest, something this section desperately needs is to have its procedures organized into numbered steps, rather than the current discursive paragraphs. I know you're inheriting this file from the previous writer, and your additions follow the pre-existing style nicely, so if you don't have time for that kind of reorganization work, no worries. (But if you do, that would be great!)

File: en_us/developers/source/conventions/internationalization/i18n.rst (hunk 6 above)
Comment (lamagnifica) [inline en_us/developers/source/conventions/internationalization/i18n.rst:416]: 😃

**Comment by lamagnifica** [comment]
👍

**Comment by staubina** [comment]
@lamagnifica 
I do not currently have time to make those changes, sorry.  I will keep it in mind as I make future updates.

**Comment by staubina** [comment]
@catong @lamagnifica 

Should I put anyone on this PR for Partner or PM review?

**Comment by lamagnifica** [comment]
@staubina it's not necessary for this change. Thanks!

**Comment by cahrens** [comment]
👍

**Comment by staubina** [comment]
@sstack22 Please take a look at these updates when you have a chance.

**Pull request merged**